2048-game.js

```diff
@@ -0,0 +1,777 @@
+                        counter++;
+                    }
+                    if (arr[edgeNo + 4].style.backgroundColor != "" && arr[edgeNo + 4].innerHTML == arr[edgeNo].innerHTML){
+                        counter++;
+                    } 
+                })
+                    
+            }
+            if(i != 0 && i != 1 && i != 2 && i != 3){
+                if (arr[i-4].style.backgroundColor == "" || arr[i-4].innerHTML == arr[i].innerHTML){
+                        counter++;
+                        }  
+                }
+        }
+    }
+    //console.log(`Up counter: ${counter}`)
+    if (counter != 0) return true;
+    else return false;
+}
+
+function canItMoveDown(){
+    let counter = 0;
+    for (i = arr.length-1; i >=0; i--){
+        if (arr[i].style.backgroundColor != ""){
+            if (i == 12 || i == 13 || i == 14 || i == 15){
+                let edges = [12,13,14,15];
+                edges.forEach((edgeNo) => {
+                    if (arr[edgeNo - 12].style.backgroundColor != "" && arr[edgeNo - 12].innerHTML == arr[edgeNo - 8].innerHTML){
+                        counter++;
+                    } 
+                    if (arr[edgeNo - 12].style.backgroundColor != "" && arr[edgeNo - 12].innerHTML == arr[edgeNo - 4].innerHTML && arr[edgeNo - 8].style.backgroundColor == "") {
+                        counter++;
+                    }
+                    if (arr[edgeNo - 12].style.backgroundColor != "" && arr[edgeNo - 12].innerHTML == arr[edgeNo].innerHTML && arr[edgeNo -  8].style.backgroundColor == "" && arr[edgeNo - 4].style.backgroundColor == "") {
+                        counter++;
+                    }
+                    if (arr[edgeNo - 8].style.backgroundColor != "" && arr[edgeNo - 8].innerHTML == arr[edgeNo - 4].innerHTML){
+                        counter++;
+                    } 
+                    if (arr[edgeNo - 8].style.backgroundColor != "" && arr[edgeNo - 8].innerHTML == arr[edgeNo].innerHTML && arr[edgeNo - 4].style.backgroundColor == "") {
+                        counter++;
+                    }
+                    if (arr[edgeNo - 4].style.backgroundColor != "" && arr[edgeNo - 4].innerHTML == arr[edgeNo].innerHTML){
+                        counter++;
+                    } 
+                })
+                    
+            }
+            if(i != 12 && i != 13 && i != 14 && i != 15){
+                if (arr[i+4].style.backgroundColor == "" || arr[i+4].innerHTML == arr[i].innerHTML){
+                        counter++;
+                        }  
+                }
+        }
+    }
+    //console.log(`Down counter: ${counter}`)
+    if (counter != 0) return true;
+    else return false;
+}
+
+//Function to see which number is in selected tile
+function numInTile(x){
+    let n = 0;
+    while (x != tiles[n].numCode){
+        n++
+    }
+    return n
+}
+
+//Function to move right
```

Comment:
Same here, but this shiftRight name is a bit vague, so use a longer name or have a better comment, i prefer having a better/longer function name

Reply:
Should explain the logic of the function on top of the pretty good comments you have on each if statement